packages/server/src/Hocuspocus.ts

```diff
@@ -414,15 +416,23 @@ export class Hocuspocus {
   /**
    * Create a new document by the given request
    */
-  public async createDocument(documentName: string, request: Partial<Pick<IncomingMessage, 'headers' | 'url'>>, socketId: string, connection: ConnectionConfiguration, context?: any): Promise<Document> {
-    if (this.documents.has(documentName)) {
-      const document = this.documents.get(documentName)
+  public createDocument(documentName: string, request: Partial<Pick<IncomingMessage, 'headers' | 'url'>>, socketId: string, connection: ConnectionConfiguration, context?: any): Promise<Document> {
+    if (this.loadingDocuments.has(documentName)) {
+      const documentPromise = this.loadingDocuments.get(documentName)
 
-      if (document) {
-        return document
+      if (documentPromise) {
+        return documentPromise
       }
     }
```

Comment:
This could be simplified
```suggestion
    if (this.loadingDocuments.has(documentName)) {
      return this.loadingDocuments.get(documentName)
    }
 ```

Reply:
This is done for TS / eslint dodging.  `has()` doesn't gaurantee `.get()` will return the value in the map since it could be nullable.  Adding a `!` seems to be prevented by eslint.

Reply:
Then maybe this might suit better

```js
const loadingDocument = this.loadingDocuments.get(documentName);

if (loadingDocument) {
  return loadingDocument
}
```